test/lib/bamboo/adapters/test_adapter_test.exs

```diff
@@ -157,4 +159,16 @@ defmodule Bamboo.TestAdapterTest do
 
     assert_delivered_email sent_email
   end
+
+  # The actual sent email and the email tested may have differences in their assigns
+  # Even if the resulting body is the same
+  test "assertion helpers remove assigns" do
+    test_email = new_email(from: "foo@bar.com", to: "bar@baz.com")
+    sent_email = %{test_email | assigns: %{foo: :bar}}
+    expected_email = %{test_email | assigns: %{foo: :baz}}
+
+    sent_email |> TestMailer.deliver_now
+
+    assert_delivered_email expected_email
```

Comment:
I think this test could be simplified a bit by doing something like

```elixir
email = new_email(from: "foo@bar.com", to: "bar@baz.com", assigns: :anything)

email |> TestMailer.deliver_now

assert_delivered_email %{email | assigns: :assigns_removed_for_testing}
```

That way the assertion more clearly shows the expected result. What do you think?